doc/_static/js/doxygen-tooltip.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2024, Benjamin Cabé <benjamin@zephyrproject.org>
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This script adds mouse hover hooks to the current page to display
+ * Doxygen documentation as tooltips when hovering over a symbol in the
+ * documentation.
+ */
+
+registerDoxygenTooltip = function () {
+  const parser = new DOMParser();
+
+  const tooltip = document.createElement('div');
+  tooltip.id = 'doxygen-tooltip';
+  document.body.appendChild(tooltip);
+
+  const tooltipShadow = document.createElement('div');
+  tooltipShadow.id = 'doxygen-tooltip-shadow';
+  tooltip.attachShadow({ mode: 'open' }).appendChild(tooltipShadow);
+
+  let style = document.createElement('style');
+
+  style.textContent += `.tippy-box {`;
```

Comment:
is there any reason to not serve this in a regular css file?

Reply:
good point, no particular reason besides that it eventually needs to be attached to the shadow root, not the main document. It's now served as a regular static resource but some gymnastics is needed to pass it to the shadow root